Packs/SalesforceIndicators/Integrations/SalesforceIndicators/SalesforceIndicators.py

```diff
@@ -0,0 +1,190 @@
+                'Content-Type': 'application/x-www-form-urlencoded'
+            },
+            data=body
+        )
+        return res
+
+    def query_object(self, fields, table, condition=None):
+        if condition:
+            query = f"SELECT {','.join(fields)} FROM {table} WHERE {condition}"
+        else:
+            query = f"SELECT {','.join(fields)} FROM {table}"
+        return self.raw_query(query)
+
+    def raw_query(self, query, raw_query=False):
+
+        params = {
+            "q": query
+        }
+
+        if raw_query:
+            res = self._http_request(
+                'GET',
+                f'{query}'
+            )
+        else:
+            res = self._http_request(
+                'GET',
+                'query',
+                params=params
+            )
+        return res
+
+    def get_object_description(self):
+        res = self._http_request('GET', f'sobjects/{self.object_name}/describe/')
+        return res
+
+
+def fetch_indicators_command(client, params, manual_run=False):
+
+    indicators_unparsed = list()
+    indicators = list()
+    now = datetime.utcnow()
+    create_date = dateparser.parse(f"{client.history} days ago", settings={
+                                   'RELATIVE_BASE': now}).strftime("%Y-%m-%dT%H:%M:%S.000+0000")
+    integration_context = get_integration_context()
+    object_fields = None
+    if client.fields:
+        object_fields = client.fields.split(",")
+    else:
+        object_fields = sorted([x['name'] for x in client.get_object_description()['fields']])
+    if "id" not in object_fields and "Id" not in object_fields:
+        object_fields.append("id")
+
+    search_criteria = str()
+    if client.query_filter:
+        search_criteria = f"{client.query_filter}"
+    else:
+        search_criteria = f"CreatedDate > {create_date}"
+    indicators_raw = client.query_object(object_fields, client.object_name, search_criteria)
+    if indicators_raw.get('totalSize', 0) > 0:
+        for indicator in indicators_raw.get('records', []):
+            indicators_unparsed.append({k: v for k, v in indicator.items() if k != 'attributes'})
+        more_records = True if indicators_raw.get('nextRecordsUrl', None) else False
+
+        while more_records:
+            next_records = "/".join(indicators_raw.get('nextRecordsUrl').split("/")[-2:])
+            indicators_raw = client.raw_query(next_records, raw_query=True)
+            for indicator in indicators_raw.get('records', []):
+                indicators_unparsed.append({k: v for k, v in indicator.items() if k != 'attributes'})
+
+            more_records = True if indicators_raw.get('nextRecordsUrl', None) else False
+
+    for item in indicators_unparsed:
+        try:
+            value = item[client.key_field] if client.key_field in item else None
+            item['object_name'] = client.object_name
+            indicator = {
+                "value": value,
+                "type": client.object_name,
```

Comment:
Since this integration receives it as a parameter, we will need to verify that it is actually a valid indicator type in the system.
This can be done by using the `FeedIndicatorType.is_valid_type` method from CommonServerPython.

Reply:
I will leave this out. This only validates that the indicator is a valid system default type and does not check if the indicator is a custom type?

Reply:
You're right in this case, resolving.